src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,279 @@
+                }
+
+                var currentNode = node;
+
+                if (needsRename)
+                {
+                    currentNode = ((SimpleNameSyntax)currentNode).WithIdentifier(identifierToken);
+                }
+
+                if (hasAdditionalTypeArguments)
+                {
+                    var existingTypeArguments = symbol.TypeArguments.Select(x => x.GenerateTypeSyntax());
+                    // Prepend additional type arguments to preserve lexical order in which they are defined
+                    var typeArguments = additionalTypeArguments.Concat(existingTypeArguments);
+                    currentNode = generator.WithTypeArguments(currentNode, typeArguments);
+                }
+
+                if (node.Parent.IsKind(SyntaxKind.InvocationExpression, out InvocationExpressionSyntax invocation))
+                {
+                    if (hasAdditionalArguments)
+                    {
+                        var shouldUseNamedArguments =
+                            !supportsNonTrailing && invocation.ArgumentList.Arguments.Any(arg => arg.NameColon != null);
+
+                        var additionalArguments = capturesAsParameters.Select(parameter =>
+                            (ArgumentSyntax)generator.Argument(
+                                name: shouldUseNamedArguments ? parameter.Name : null,
+                                refKind: parameter.RefKind,
+                                expression: parameter.Name.ToIdentifierName())).ToArray();
+
+                        editor.ReplaceNode(invocation.ArgumentList,
+                            invocation.ArgumentList.AddArguments(additionalArguments));
+                    }
+                }
+                else if (hasAdditionalArguments || hasAdditionalTypeArguments)
+                {
+                    // Convert local function delegates to lambda if the signature no longer matches
+                    var arguments = parameters
+                        .Concat(capturesAsParameters)
+                        .Select(parameter => generator.Argument(
+                            name: null, refKind: parameter.RefKind,
+                            expression: parameter.Name.ToIdentifierName()));
+
+                    currentNode = generator.ValueReturningLambdaExpression(
+                        lambdaParameters: ParameterGenerator.GetParameters(parameters, isExplicit: true, options: defaultOptions),
+                        expression: generator.InvocationExpression(currentNode, arguments));
+
+                    if (node.IsParentKind(SyntaxKind.CastExpression))
+                    {
+                        currentNode = ((ExpressionSyntax)currentNode).Parenthesize();
+                    }
+                }
+
+                editor.ReplaceNode(node, currentNode);
+            }
+
+            root = editor.GetChangedRoot();
+            localFunction = (LocalFunctionStatementSyntax)root.FindNode(localFunction.Span);
```

Comment:
risky.  but i suppose this would work.  in general, a cleaner way to do this is to add an annotation to the node, and use that annotatoin to find the node after the change.  This means you're resilient to any spans being changed.   I know that *shouldn't* happen since you're adding *after*... but strange things do happen.